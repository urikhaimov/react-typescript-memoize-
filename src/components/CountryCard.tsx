import React from 'react';
import Popup from 'reactjs-popup';
// import  'reactjs-popup/dist/index.css';
import { Country } from '../types/Country';
// import defaultFlagIcon from '../resources/images/default.svg'
export interface Props {
  country: Country
  clickHandler(c: Country): void;
}

const CountryCard: React.FC<Props> = ({country, clickHandler }) => {
  const name = country.name.official;
  const flag = country.flags.svg ;
return (
    <div className="country">
      <h2>{name}</h2>
      <Popup trigger={<img src={flag} alt={name} />} position="right center">
        <div>
          <img src={flag} alt={name} onClick={clickHandler.bind(null,country)} />
        </div>
      </Popup>

    </div>
  );
};

export default CountryCard;
