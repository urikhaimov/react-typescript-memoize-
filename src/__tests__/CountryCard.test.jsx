import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup, fireEvent, render } from '@testing-library/react';
import CountryCard from '../components/CountryCard';


const addCountryToStorage = jest.fn()
const country = {
  name: {
    common: 'WonderLand',
    official: 'WonderLand'
  },
  flags: {
    png: 'png',
    svg: 'svg'
  },
  idd: {
    root: '+5',
    suffixes: [0]
  }

}

afterEach(cleanup);

test('render Country Card', () => {

  const component = renderer.create(
    <CountryCard
      key={`countryCard-${0}`}
      country={country}
      clickHandler={addCountryToStorage}
    />,

  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0]).toEqual('WonderLand');
  expect(tree.children[1].props.src).toEqual('svg');


});

it('Country Card after click', () => {
  const { queryAllByRole, getAllByRole } = render(
    <CountryCard key={`countryCard-${0}`}
      country={country}
      clickHandler={addCountryToStorage} />,
  );
  let img = queryAllByRole('img')
  expect(img[0]).toBeTruthy();
  expect(img.length).toEqual(1);
  fireEvent.click(getAllByRole('img')[0]);
  img = queryAllByRole('img');
  expect(img.length).toEqual(2);
});