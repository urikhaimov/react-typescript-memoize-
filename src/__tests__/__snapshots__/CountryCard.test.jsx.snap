// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Country Card 1`] = `
<div
  className="country"
>
  <h2>
    WonderLand
  </h2>
  <img
    alt="WonderLand"
    aria-describedby="popup-1"
    id="small-image-WonderLand"
    onClick={[Function]}
    src="svg"
  />
</div>
`;
